tools/wpt/testharnessreport.js

```diff
@@ -1,13 +1,15 @@
 // Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
 window.add_result_callback(({ message, name, stack, status }) => {
-  Deno.writeAllSync(
-    Deno.stderr,
-    new TextEncoder().encode(
-      `${JSON.stringify({ name, status, message, stack })}\n`,
-    ),
+  const data = new TextEncoder().encode(
+    `${JSON.stringify({ name, status, message, stack })}\n`,
   );
+  const bytesWritten = Deno.stderr.writeSync(data);
+  if (bytesWritten !== data.byteLength) {
+    throw new TypeError("failed to report test result");
+  }
```

Comment:
Shouldn't we make sure that all data have been written to stderr like the following?

```suggestion
  let bytesWritten = 0;
  while (bytesWritten < data.byteLength) {
    bytesWritten += Deno.stderr.writeSync(data.subarray(byteWritten));
  }
```

Reply:
that works too :-)

Reply:
so either is okay? then I'm fine with the current implementation of yours.
I feel a little bit skeptical though, because `Deno.writeSync` calls `op_write_sync` and then `std::io::Write::write` is used under the hood, which requires us to do a loop until all data are written if we wish to imitate something like write_all.
https://doc.rust-lang.org/std/io/trait.Write.html#examples

Reply:
I'm aware, I just think the messages are small enough to never fail. But if you want to add the loop, that is fine by me. It is a little more robust.

Reply:
Makes sense. Indeed the tests run in CI succeeded, but I'd prefer a robust one so I'll add the loop :)